### PR TITLE
fix(credential-ld): fix Ed25519Signature2020 verification

### PR DESCRIPTION
**Pull request by mirceanis**
## What issue is this PR fixing

This fixes the Ed25519Signature2020 signature verification.

## What is being changed

This PR fixes several issues with the suite:
* The signature encoding was broken (it was interpreted as utf-8 instead of base64url)
* The underlying LD-suite was expecting the document loader to also dereference DID URLs
* The other suites would still expect full DID documents to be returned by the documentLoader
* The underlying signature suite expected a particular `@context` and `publicKeyMultibase` to be set on the verification method used for verification

This PR also cleans the default contexts and adds a `schema.org` `@context` as well.

## Quality
Check all that apply:
* [X] I want these changes to be integrated
* [X] I successfully ran `pnpm i`, `pnpm build`, `pnpm test`, `pnpm test:browser` locally.
* [X] I allow my PR to be updated by the reviewers (to speed up the review process).
* [X] I added unit tests.
* [X] I added integration tests.

closes #1146

### CODE REVIEW

File: packages/credential-ld/src/suites/Ed25519Signature2020.ts [hunk 1]
```diff
@@ -40,15 +51,16 @@ export class VeramoEd25519Signature2020 extends VeramoLdSignature {
           keyRef: key.kid,
           data: messageString,
           encoding: 'base64',
+          algorithm: 'EdDSA',
         })
-        return u8a.fromString(signature)
+        return base64ToBytes(signature)
```
Comment (mirceanis) [inline packages/credential-ld/src/suites/Ed25519Signature2020.ts:56]: This was the culprit of signature verification failures

File: packages/credential-ld/src/suites/Ed25519Signature2020.ts [hunk 2]
```diff
@@ -69,12 +81,61 @@ export class VeramoEd25519Signature2020 extends VeramoLdSignature {
...
+    let document = doc
+    // The verification method (key) must contain "https://w3id.org/security/suites/ed25519-2020/v1" context.
+    if ((document as DIDDocument).verificationMethod) {
+      document.verificationMethod = asArray(document.verificationMethod)?.map(
+        this.transformVerificationMethod,
+      )
+    }
+
+    // this signature suite requires the document loader to dereference the DID URL
+    if (didUrl.includes('#') && didUrl !== document.id) {
```
Comment (mirceanis) [inline packages/credential-ld/src/suites/Ed25519Signature2020.ts:98]: This is the workaround for the very strict documentLoader requirements.

**Review by mirceanis — COMMENTED** [review]
some comments to help the review

File: packages/credential-ld/src/ld-credential-module.ts [hunk 3]
```diff
@@ -46,16 +50,20 @@ export class LdCredentialModule {
 
         if (!didDoc) return
 
+        let result: any = didDoc
```
Comment (nickreynolds) [inline packages/credential-ld/src/ld-credential-module.ts:53]: this needs to be `any`?

**Review by nickreynolds — APPROVED** [review]

File: packages/credential-ld/src/ld-credential-module.ts (hunk 3 above)
Comment (mirceanis) [reply packages/credential-ld/src/ld-credential-module.ts:53]: Yeah, because later it can become a verification method instead of a did doc

**Comment by Eengineer1** [comment]
Nice work getting to the bottom of this!

**Pull request merged**